public/app/features/explore/RichHistory/RichHistoryContainer.test.tsx

```diff
@@ -16,26 +15,31 @@ const setup = (propOverrides?: Partial<Props>) => {
     richHistory: [],
     firstTab: Tabs.RichHistory,
     deleteRichHistory: jest.fn(),
+    loadRichHistory: jest.fn(),
     onClose: jest.fn(),
   };
 
   Object.assign(props, propOverrides);
 
-  const wrapper = mount(<RichHistoryContainer {...props} />);
-  return wrapper;
+  return render(<RichHistoryContainer {...props} />);
 };
 
 describe('RichHistoryContainer', () => {
-  it('should render reseizable component', () => {
-    const wrapper = setup();
-    expect(wrapper.find(Resizable)).toHaveLength(1);
-  });
   it('should render component with correct width', () => {
-    const wrapper = setup();
-    expect(wrapper.getDOMNode().getAttribute('style')).toContain('width: 531.5px');
+    const { container } = setup();
+    expect(container.firstElementChild!.getAttribute('style')).toContain('width: 531.5px');
   });
   it('should render component with correct height', () => {
-    const wrapper = setup();
-    expect(wrapper.getDOMNode().getAttribute('style')).toContain('height: 400px');
+    const { container } = setup();
+    expect(container.firstElementChild!.getAttribute('style')).toContain('height: 400px');
+  });
+  it('should request loading rich history when mounted', () => {
+    const loadRichHistory = jest.fn();
+    const { unmount } = setup({ loadRichHistory });
+    expect(loadRichHistory).toBeCalledTimes(1);
+
+    unmount();
+    setup({ loadRichHistory });
+    expect(loadRichHistory).toBeCalledTimes(2);
```

Comment:
should it really also re-request loading rich history when unmounting? if so, can we update the test name?

Reply:
It should only re-request when mounted. I've added extra assertion after unmount and updated the name in 1ddb359ebcef95d051ca2303c38ccb895c328284.